cpp/include/cudf/utilities/span.hpp

```diff
@@ -350,7 +358,9 @@ struct device_span : public cudf::detail::span_base<T, Extent, device_span<T, Ex
                                std::remove_pointer_t<decltype(thrust::raw_pointer_cast(  // NOLINT
                                  std::declval<C&>().data()))> (*)[],
                                T (*)[]>>* = nullptr>  // NOLINT
-  constexpr device_span(C& in) : base(thrust::raw_pointer_cast(in.data()), in.size())
+  CUDF_HOST_DEVICE constexpr device_span(C& in) : base()
+  // CUDF_HOST_DEVICE constexpr device_span(C& in) : base(thrust::raw_pointer_cast(in.data()),
+  // in.size())
```

Comment:
What is happening here?

Reply:
Sorry this is totally a goof on my part. I was running into errors in an earlier iteration of this PR and this was just trying to unblock my progress. It should not have made it into this PR.

Reply:
For context, I have to revert changing this function to `CUDF_HOST_DEVICE` because that doesn't actually compile. I couldn't tell why, it seemed like the compiler was selecting the incorrect overload of the base constructor both here and in the similar line above and it didn't seem correct. I'm just going to revert this change for the moment and we can revisit in a future round of changes.